pyrefly/lib/lsp/wasm/hover.rs

```diff
@@ -236,6 +248,60 @@ fn expand_callable_kwargs_for_hover<'a>(
         }
     }
 }
+
+fn fallback_hover_name_from_type(type_: &Type, current_module: ModuleName) -> Option<String> {
+    match type_ {
+        Type::Function(function) => Some(function.metadata.kind.format(current_module)),
+        Type::BoundMethod(bound_method) => match &bound_method.func {
+            BoundMethodType::Function(function) => {
+                Some(function.metadata.kind.format(current_module))
+            }
+            BoundMethodType::Forall(forall) => {
+                Some(forall.body.metadata.kind.format(current_module))
+            }
+            BoundMethodType::Overload(overload) => {
+                Some(overload.metadata.kind.format(current_module))
+            }
+        },
+        Type::Overload(overload) => Some(overload.metadata.kind.format(current_module)),
+        Type::Forall(forall) => match &forall.body {
+            Forallable::Function(function) => Some(function.metadata.kind.format(current_module)),
+            Forallable::Callable(_) | Forallable::TypeAlias(_) => None,
+        },
+        Type::Type(inner) => fallback_hover_name_from_type(inner, current_module),
+        _ => None,
+    }
+}
+
+fn identifier_text_at(
+    transaction: &Transaction<'_>,
+    handle: &Handle,
+    position: TextSize,
+) -> Option<String> {
+    let module = transaction.get_module_info(handle)?;
```

Comment:
we have a `Module::code_at` function already. can we reuse that and avoid `identifier_text_at` entirely?